sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorEventSource.cs

```diff
@@ -0,0 +1,395 @@
+        ///   Provides a singleton instance of the event source for callers to
+        ///   use for logging.
+        /// </summary>
+        ///
+        public static EventProcessorEventSource Log { get; } = new EventProcessorEventSource();
+
+        internal EventProcessorEventSource() { }
+
+        /// <summary>
+        ///   Signals the <see cref="EventProcessorClient" /> to begin processing events.
+        /// </summary>
+        ///
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        ///
+        [Event(1, Level = EventLevel.Informational, Message = "Starting a new event processor instance with identifier '{0}'.")]
+        public virtual void EventProcessorStart(string identifier)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(1, identifier);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the process of load balancing has started.
+        /// </summary>
+        [Event(2, Level = EventLevel.Informational, Message = "The process of balancing load between event processors has started")]
+        public void StartLoadBalacing()
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(2);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates the total number of ownership records returned by storage manager.
+        /// </summary>
+        ///
+        /// <param name="count">Total number of ownership records returned by storage manager.</param>
+        ///
+        [Event(3, Level = EventLevel.Informational, Message = "Storage manager returned '{0}' ownership records.")]
+        public virtual void CompleteOwnershipRecordsCount(int count)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(3, count);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while retrieving ownership records.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace that the processor is associated with.</param>
+        /// <param name="eventHubName">The name of the Event Hub that the processor is connected to, specific to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group this event processor is associated with.  Events will be read only in the context of this group.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(4, Level = EventLevel.Error, Message = "An exception occurred while retrieving ownership records. (FullyQualifiedNamespace: '{0}'; EventHubName: '{1}'; ConsumerGroup: '{2}'; ErrorMessage: '{3}')")]
+        public virtual void ListOwnershipAsyncError(string fullyQualifiedNamespace, string eventHubName, string consumerGroup, string errorMessage)
```

Comment:
I'll update this but just wondering is this a .Net format that we should follow?

Reply:
I'm pretty new to all this stuff as well, but I believe it's something we consistently do in our client rather than a .NET convention. @jsquire certainly has more context.

Reply:
There's no formal standard, but it is helpful for consistency and readability.   There have been studies done where the outcome is that breaking parameters onto their own lines helps to reduce mistakes.   Likewise with aligning things, but I lost my push for that early on.  😄 

So, like the "we include document comments on every member because we want those working on the code to have the benefit of context",  the parameter-per-line is currently just a convention within Event Hubs.